src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -1436,6 +1408,36 @@ int wm_vuldet_process_agent_vulnerabilities(sqlite3 *db, OSHash *cve_table, scan
                 }
             }
 
+            //Save the vulnerability in the agent database
+            bool check_pkg_existence = !strcmp(pkg->type, VULN_CVES_TYPE_PACKAGE);
+            cJSON* j_result = wdb_insert_vuln_cves(scan_ctx->agent_id, pkg->bin_name, pkg->version, pkg->arch, cve,
```

Comment:
This is not related to your implementation, but I discovered that we have a potential issue with the string of the version that we are using. If you take a look at the piece of code that fills the `report->version` field, it does something like this...

```c
            if (!strncmp("-1:", pkg->version, 3)) {
                w_strdup((pkg->version + 3), report->version);
            } else {
                w_strdup(pkg->version, report->version);
            }
```

So if we make use of `pkg->version`, we will not be handling the case of versions starting with `-1:`. The result will be a difference between the versions in the alerts and the inventory.

My suggestion here is to change  `pkg->bin_name` with `report->software`, also `pkg->version` with `report->version`, and `pkg->arch` with `report->arch`. So in this way we make sure that the inventory will have exactly the same information reported in the alerts.

Reply:
Indeed is related to these changes, I did move that block of code. Good catch, thanks